internal/collector/storageclass.go

```diff
@@ -0,0 +1,122 @@
+/*
+Copyright 2019 The Kubernetes Authors All rights reserved.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package collector
+
+import (
+	"k8s.io/kube-state-metrics/pkg/metric"
+
+	storagev1 "k8s.io/api/storage/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+)
+
+var (
+	descStorageClassLabelsName          = "kube_storageclass_labels"
+	descStorageClassLabelsHelp          = "Kubernetes labels converted to Prometheus labels."
+	descStorageClassLabelsDefaultLabels = []string{"storageclass"}
+
+	storageClassMetricFamilies = []metric.FamilyGenerator{
+		{
+			Name: "kube_storageclass_info",
+			Type: metric.Gauge,
+			Help: "Information about storageclass.",
+			GenerateFunc: wrapStorageClassFunc(func(s *storagev1.StorageClass) *metric.Family {
+				m := metric.Metric{
+					LabelKeys:   []string{"provisioner", "reclaimPolicy", "volumeBindingMode"},
+					LabelValues: []string{s.Provisioner, string(*s.ReclaimPolicy), string(*s.VolumeBindingMode)},
+					Value:       1,
+				}
+				return &metric.Family{Metrics: []*metric.Metric{&m}}
+			}),
+		},
+		{
+			Name: "kube_storageclass_created",
+			Type: metric.Gauge,
+			Help: "Unix creation timestamp",
+			GenerateFunc: wrapStorageClassFunc(func(s *storagev1.StorageClass) *metric.Family {
+				ms := []*metric.Metric{}
+				if !s.CreationTimestamp.IsZero() {
+					ms = append(ms, &metric.Metric{
+						Value: float64(s.CreationTimestamp.Unix()),
+					})
+				}
+				return &metric.Family{
+					Metrics: ms,
+				}
+			}),
+		},
+		{
+			Name: "kube_storageclass_metadata_resource_version",
+			Type: metric.Gauge,
+			Help: "Resource version representing a specific version of storageclass.",
+			GenerateFunc: wrapStorageClassFunc(func(s *storagev1.StorageClass) *metric.Family {
+				return &metric.Family{
+					Metrics: []*metric.Metric{
+						{
+							LabelKeys:   []string{"resource_version"},
+							LabelValues: []string{s.ObjectMeta.ResourceVersion},
```

Comment:
I'm still a little sceptic about this. If at all, why wouldn't this be the metric's value?

Reply:
I refer to other code which use `ResourceVersion` as metric, I found all of them use `1` as metric's value. 
for example: `kube_secret_metadata_resource_version`, `kube_configmap_metadata_resource_version` and `kube_ingress_metadata_resource_version`.

Reply:
Wow, you're totally right the other metrics do that, but it's wrong, we should have never merged those. Thanks for noticing! We should add these to the breaking release issue.

Reply:
I think we should skip this one or model it in a different way for now. Do you have a concrete alerting use case in mind?

Reply:
I remove `resource_version` for the time being, and I think it's not necessary to alert about `storageclass`, I only want to monitor its info.